app/models/user.rb

```diff
@@ -4,14 +4,20 @@ class User < ActiveRecord::Base
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable, :confirmable
 
-  has_many :posts, foreign_key: 'author_id', class_name: 'Post'
-  has_many :likes, foreign_key: 'author_id', class_name: 'Like'
-  has_many :comments, foreign_key: 'author_id', class_name: 'Comment'
+  has_many :posts, foreign_key: 'author_id', class_name: 'Post', dependent: :delete_all
+  has_many :likes, foreign_key: 'author_id', class_name: 'Like', dependent: :delete_all
+  has_many :comments, foreign_key: 'author_id', class_name: 'Comment', dependent: :delete_all
 
-  # validates :name, presence: true
-  # validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   def most_recent_posts
     posts.limit(3).order(created_at: :desc)
   end
+
+  def admin?
+    return true if role == 'admin'
+
+    false
+  end
```

Comment:
- [ ] God job adding the `admin` method :wink:

Reply:
Thank you :smile: